#     E------F
#    /|     /|
#   A------B |
#   | |    | |
#   | G----|-H
#   |/     |/
#   C------D
point_2d A
point_2d B
point_2d C
point_2d D
point_2d E
point_2d F
point_2d G
point_2d H

point_2d center

Vector2 speed